src/main/java/com/example/hop_oasis/service/data/CartServiceImpl.java

```diff
@@ -154,12 +155,15 @@ private CartItemDto createCartItemDto(CartItem cartItem) {
             }
             default -> throw new ResourceNotFoundException(RESOURCE_NOT_FOUND, "");
         }
-        BigDecimal roundedTotalCost = BigDecimal.valueOf(dto.getQuantity() * dto.getPricePerItem())
-                .setScale(2, RoundingMode.HALF_UP);
-        dto.setTotalCost(roundedTotalCost);
+        dto.setTotalCost(roundTotalCost(dto.getQuantity(), dto.getPricePerItem()));
```

Comment:
consider making util class
https://mammadyahya.medium.com/how-to-design-utility-classes-in-java-15772c5a6e1f

that will be responsible for rouding.
you can create some  `round(double num)` mehtod

in this case you can do it like:
```
        dto.setTotalCost(YourUtilClass.round(dto.getQuantity() * dto.getPricePerItem()));
```

Reply:
done